File: exercism/c/archive/hello-world/src/hello_world.c
#include <stdio.h>
#include <stddef.h>
#include <string.h>

#include "hello_world.h"

void hello(char *buffer, const char *name) {
    if (name == NULL) {
        sprintf(buffer, "%s", "Hello, World!");
        return;
    }

    sprintf(buffer, "%s, %s!", "Hello", name);
}

File: exercism/go/archive/hello-world/hello_world.go
// Package hello gives us our first foray into golang exercism content
package hello

import "fmt"

// testVersion denotes which test we're runnign
const testVersion = 2

// HelloWorld is a function name to return a nice greeting to the user!
func HelloWorld(name string) string {
	if name != "" {
		return fmt.Sprintf("Hello, %s!", name)
	}

	return "Hello, World!"
}
